src/main/java/frc/robot/util/ThroughBoreEncoder.java

```diff
@@ -0,0 +1,67 @@
+package frc.robot.util;
+
+import edu.wpi.first.math.geometry.Rotation2d;
+import edu.wpi.first.wpilibj.DutyCycleEncoder;
+import edu.wpi.first.wpilibj.Encoder;
+
+/**
+ * Wrapper class around the wpilib {@link DutyCycleEncoder} and {@link Encoder} classes for the REV
+ * ThroughBoreEncoder. Implements both the relative and absolute encoders of the device. <a
+ * href="https://docs.revrobotics.com/through-bore-encoder/specifications">Datasheet</a>
+ */
+public class ThroughBoreEncoder implements AutoCloseable {
+  private final DutyCycleEncoder m_absoluteEncoder;
+  private final Encoder m_relativeEncoder;
+
+  /**
+   * Create a ThroughBoreEncoder
+   *
+   * @param dutyCyclePort DIO port of the ABS (white) wire
+   * @param relativeAPort DIO port of the ENC A (blue) white
+   * @param relativeBPort DIO port of the ENC B (yellow) white
+   * @param absoluteEncoderOffset Offset of the duty cycle encoder. This value is subtracted from
+   *     the true absolute position to yield the result.
+   */
+  public ThroughBoreEncoder(
```

Comment:
nit: rename to `REVThroughBoreEncoder` as there are many types of through bore encoders